lib/wavefront-direct-reporter.js

```diff
@@ -0,0 +1,229 @@
+};
+
+WavefrontDirectReporter.prototype.appendTimer = function(timer, points) {
+  var tags = tagger(this.registry.getTags(timer.name), this.globaltags);
+  if (!tags) { tags = ''; }
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, timer.name, 'count', timer.count(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, timer.name, 'mean_rate', timer.meanRate(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, timer.name, 'm1_rate', timer.oneMinuteRate(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, timer.name, 'm5_rate', timer.fiveMinuteRate(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, timer.name, 'm15_rate', timer.fifteenMinuteRate(), tags));
+
+  this.appendHistogram(timer, points);
+};
+
+WavefrontDirectReporter.prototype.appendHistogram = function(histogram, points) {
+  var tags = tagger(this.registry.getTags(histogram.name),this.globaltags);
+  var isHisto = Object.getPrototypeOf(histogram) === Histogram.prototype;
+  if (isHisto) {
+    // send count if a histogram, otherwise assume this metric is being
+    // printed as part of another (like a timer).
+    points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'count', histogram.count, tags));
+  }
+
+  var percentiles = histogram.percentiles([.50,.75,.95,.98,.99,.999]);
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'min', isHisto? histogram.min : histogram.min(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'mean', histogram.mean(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'max', isHisto ? histogram.max: histogram.max(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'stddev', histogram.stdDev(), tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'p50', percentiles[.50], tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'p75', percentiles[.75], tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'p95', percentiles[.95], tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'p98', percentiles[.98], tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'p99', percentiles[.99], tags));
+  points.push(util.format('%s.%s.%s %s %s', this.prefix, histogram.name, 'p999', percentiles[.999], tags));
+};
+
+/**
+ * Returns an array with arrays of the given size.
+ *
+ * @param points {Array} Array to split
+ * @param chunkSize {Integer} Size of every chunk
+ */
+function getChunks(points, chunk_size){
+    var chunks = [];
+    while (points.length) {
+        chunks.push(points.splice(0, chunk_size));
+    }
+    return chunks;
+}
+
+function tagger(tags,globaltags) {
+  //return tags in the format 'env="prod" service="qa"'
```

Comment:
Note that your code does not do a real merge (union), as it does not deduplicate entries.  Is this intended?
If you want to merge the contents of two objects into another object, use the spread operator `...`

Incremental concatenation of strings is expensive, the accepted way is to build an array of strings and use Array.join().

```javascript
mergedTags = {...tags, ...globaltags};
return Object.keys(mergedTags).map(key => `${key}="${escapeQuotes(mergedTags[key])}"`).join(' ');
```